src/collective.jl

```diff
@@ -39,6 +39,19 @@ function Bcast!(data, root::Integer, comm::Comm)
     Bcast!(Buffer(data), root, comm)
 end
 
+"""
+    Bcast(obj::T, root::Integer, comm::Comm) where T
+
+Broadcast the `obj` from `root` to all processes in `comm`. Returns the object.
+"""
+function Bcast(obj::T, root::Integer, comm::Comm) where T
+    buf = Ref{T}(obj)
+    Bcast!(buf, root, comm)
+    return buf[]
+end
```

Comment:
This ends up being a one-liner
```suggestion
Bcast(obj, root::Integer, comm::Comm) = Bcast!(Ref(obj), root, comm)[]
```
I'm still not sure it is worth it?